app/scripts/views/complete_sign_up.js

```diff
@@ -60,6 +63,22 @@ function (Cocktail, FormView, BaseView, CompleteSignUpTemplate,
       return self.fxaClient.verifyCode(uid, code)
           .then(function () {
             self.logScreenEvent('verification.success');
+            var account = self.getAccount();
+
+            if (account.get('needsOptedInToMarketingEmail')) {
+              account.unset('needsOptedInToMarketingEmail');
+              // TODO is the account automatically persisted?
+
+              var emailPrefs = account.getMarketingEmailPrefs();
+              return emailPrefs.optIn(NEWSLETTER_ID)
```

Comment:
I'd vote for just tossing it over the wall and crossing our fingers.
I'm on a lot of mailing lists (mostly via mailchimp, oddly) and I don't remember any newsletters beach-balling my browser while the server adds me to the list.